### ISSUE TITLE
extensions – incorrectly flagging internal package imports

### ISSUE THREAD

**Issue by TehShrike** [benmosher/eslint-plugin-import#1887]
With an eslintrc containing this rule configuration: `'import/extensions': [ 1, 'ignorePackages' ]`

a file containing an [internal package import](https://nodejs.org/dist/latest-v14.x/docs/api/esm.html#esm_internal_package_imports):

```js
import dep from '#dep'
```

is incorrectly being flagged as a violation of the rule.

> warning  Missing file extension for "#dep"  import/extensions

**Comment by ljharb** [comment]
Indeed; these aren't supported until the `resolve` package supports them.
